sherpa/astro/data.py

```diff
@@ -4494,50 +4529,60 @@ def _world_to_physical(self, x0=None, x1=None):
 
         return (x0, x1)
 
+    # Convert from the _orig_indep_axis tuple (coord, x0, x1) to the
+    # required data system (if it isn't already set).
+    #
+    def _get_coordsys(self, coord):
+        if self.coord == coord:
+            return self.get_indep()
+
+        (base, x0, x1) = self._orig_indep_axis
+        x0 = x0.copy()
+        x1 = x1.copy()
+        if base == coord:
+            return (x0, x1)
+
+        conv = getattr(self, f'_{base}_to_{coord}')
+        return conv(x0, x1)
+
     def get_logical(self):
-        coord = self.coord
-        x0, x1 = self.get_indep()
-        if coord != 'logical':
-            x0 = x0.copy()
-            x1 = x1.copy()
-            x0, x1 = getattr(self, '_' + coord + '_to_logical')(x0, x1)
-        return (x0, x1)
+        return self._get_coordsys("logical")
 
     def get_physical(self):
-        coord = self.coord
-        x0, x1 = self.get_indep()
-        if coord != 'physical':
-            x0 = x0.copy()
-            x1 = x1.copy()
-            x0, x1 = getattr(self, '_' + coord + '_to_physical')(x0, x1)
-        return (x0, x1)
+        return self._get_coordsys("physical")
 
     def get_world(self):
-        coord = self.coord
-        x0, x1 = self.get_indep()
-        if coord != 'world':
-            x0 = x0.copy()
-            x1 = x1.copy()
-            x0, x1 = getattr(self, '_' + coord + '_to_world')(x0, x1)
-        return (x0, x1)
+        return self._get_coordsys("world")
 
     # For compatibility with old Sherpa keywords
     get_image = get_logical
     get_wcs = get_world
 
     def set_coord(self, coord):
+        """Change the `coord` attribute.
+
+        Parameters
+        ----------
+        coord : {'logical', 'image', 'physical', 'world', 'wcs'}
+            The coordinate system to use. Note that "image" is a
+            synomym for "logical" and "wcs" is a synomyn for "world".
+
+        """
+
         coord = str(coord).strip().lower()
-        # Destroys original data to conserve memory for big imgs
         good = ('logical', 'image', 'physical', 'world', 'wcs')
         if coord not in good:
             raise DataErr('badchoices', 'coordinates', coord, ", ".join(good))
 
-        if coord.startswith('wcs'):
+        if coord == 'wcs':
             coord = 'world'
-        elif coord.startswith('image'):
+        elif coord == 'image':
```

Comment:
Technically, this is an API breaking change, because there could be existing user code that used `set_coord("image_crap")`.... but I think it's good to be explicit!

Just commenting because I recently rediscovered https://xkcd.com/1172/ ;-)

Reply:
Actually, given the check just above we know that if we got here and trigger the image then the argument could not have been `"image_your_base_belong_to_us"` - see the error check on lines 4572 to 4575